Dictionary Project

Leah2356 opened the pull request:
<img width="1440" alt="Screen Shot 2024-04-10 at 3 50 09 PM" src="https://github.com/Leah2356/feiner-dictionary-2024/assets/126403626/f9877012-2a83-40c1-850f-67bc417b614c">

Leah2356 committed:
added to project

andrewoid commented on src/test/java/feiner/dictionary/EnglishDictionaryTest.java:
```diff
@@ -0,0 +1,27 @@
+        EnglishDictionary dictionary = new EnglishDictionary();
+
+        // when
+        List<String> definitions = dictionary.getDefinition("hello");
```
Test a word that has multiple definitions

andrewoid replied:
Please address this comment.

andrewoid commented on src/main/java/feiner/dictionary/GUI.java:
```diff
@@ -0,0 +1,56 @@
+            if (definitions != null && !definitions.isEmpty()) {
+                for (String d : definitions) {
+                    definitionsArea.append(d);
+                }
```
Did this look correct if you try a word with multiple definitions?

andrewoid reviewed (commented):
Add a readme

Leah2356 committed:
added a readme

Leah2356 committed:
fixed checkstyles

andrewoid commented on screenshots/Screen Shot 2024-04-10 at 2.07.59 PM.png:
For some reason, github can't render this file.

andrewoid commented on src/main/java/feiner/dictionary/Gui.java:
```diff
@@ -0,0 +1,56 @@
+        if (!word.isEmpty()) {
+            List<String> definitions = dictionary.getDefinition(word);
+            if (definitions != null && !definitions.isEmpty()) {
+                for (String d : definitions) {
+                    definitionsArea.append(d);
+                }
```
Does this look right with words that have multiple definitions?

Leah2356 commented:
I added a space between definitions, and it looks right now. Also,
regarding the screenshot, do you want me to post it in the comments because
it's not going through.
Thank you!

On Wed, Apr 10, 2024 at 2:41 PM Andrew Schwimmer ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/feiner/dictionary/Gui.java
> <https://github.com/Leah2356/feiner-dictionary-2024/pull/1#discussion_r1559908946>
> :
>
> > +                for (String d : definitions) {
> +                    definitionsArea.append(d);
> +                }
>
> Does this look right with words that have multiple definitions?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Leah2356/feiner-dictionary-2024/pull/1#pullrequestreview-1992430374>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6EMIKQ5CCQTNV3AAWUYZ53Y4WBW5AVCNFSM6AAAAABF75ECZOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSOJSGQZTAMZXGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Leah2356 committed:
fixed comments

Leah2356 committed:
fixed test

Leah2356 commented:
I addressed it, and I added the screenshot to my comments because it's not
working on readme.

On Wed, Apr 10, 2024 at 3:37 PM Andrew Schwimmer ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/test/java/feiner/dictionary/EnglishDictionaryTest.java
> <https://github.com/Leah2356/feiner-dictionary-2024/pull/1#discussion_r1559968609>
> :
>
> > +import org.junit.jupiter.api.Test;
> +
> +import java.io.IOException;
> +import java.util.List;
> +
> +import static org.junit.jupiter.api.Assertions.*;
> +
> +class EnglishDictionaryTest {
> +
> +    @Test
> +    public void getDefinition() throws CsvValidationException, IOException {
> +        // given
> +        EnglishDictionary dictionary = new EnglishDictionary();
> +
> +        // when
> +        List<String> definitions = dictionary.getDefinition("hello");
>
> Please address this comment.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Leah2356/feiner-dictionary-2024/pull/1#discussion_r1559968609>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6EMIKTY5UL3V4JVM4BJZA3Y4WIIVAVCNFSM6AAAAABF75ECZOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSOJSGUZTCMJUG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Leah2356 committed:
fixed chckstyle errors
